packages/notebook/style/base.css

```diff
@@ -137,6 +137,10 @@
   background: transparent;
 }
 
+.jp-Notebook.jp-mod-commandMode .jp-Cell:focus {
+  outline: solid var(--jp-inverse-layout-color0) 1px;
```

Comment:
Hi @fcollonval, I tried to make the cell:focus outlines a bit less heavy, what do you think?

| Light Theme | Dark Theme |
|-|-|
| <img width="1840" alt="JupyterLab light mode" src="https://github.com/jupyterlab/jupyterlab/assets/317883/e8da7466-5553-4b92-a61a-e5c9f24b6e67"> | <img width="1840" alt="JupyterLab dark mode" src="https://github.com/jupyterlab/jupyterlab/assets/317883/7f36a54a-e605-489c-82d2-e3795a226c04"> |
| #&#8203;757575 focus outline color | #&#8203;757575 focus outline color (yes, same as light theme) |
| #fff background | #&#8203;111 background |
| contrast ratio [4.6:1](https://webaim.org/resources/contrastchecker/?fcolor=757575&bcolor=FFFFFF) | contrast ratio [4.09:1](https://webaim.org/resources/contrastchecker/?fcolor=757575&bcolor=111111) |

To make the focus ring less heavy, I decreased the contrast of the outline with the background, and I also rounded the corners. 

In CSS, the changes look like this:

```suggestion
  outline: none;
  border: none;
  border-radius: 2px;
  box-shadow: 0 0 0 1px var(--jp-inverse-layout-color4);
  transition: 0.1s;
```

This idea came out of a short brainstorming session with @trallard and @steff456. We also considered the idea (brought up earlier in this PR's conversation history) of repurposing the current active cell indicator (the blue bar to the left of an active cell), but we were concerned that overloading that indicator could cause confusion (because it's not a common pattern).

If this still feels like too heavy of a UI change, then I think maybe the next best course of action is to go with @tonyfast's idea in the JupyterLab team call this week. We could punt on focus styling for now and keep working on it until we find a solution that both satisfies WCAG and yet isn't too intrusive to the existing UI, so that the rest of this PR can be reviewed for its main intended purpose, which is to fix the keyboard trap issue.

A note about `:focus-visible`... You might be wondering, if you've played with focus styles before, why not just use the `:focus-visible` selector instead of `:focus`. After all, the point of `:focus-visible` is to make a compromise between people who want the extra visual noise of a focus ring versus those who don't. Generally speaking, it's supposed to tell the browser: "hey browser, use these styles when you think the user really needs them (for example, when they can't use a mouse); otherwise don't show them."  So I tried that but I could not get consistent behavior in my browser (Chrome). The behavior I want is that if you load JupyterLab and you only interact with the UI using a mouse, no focus outline should ever appear on any notebook cell. But what I observed is that the focus ring would appear after page load when selecting via mouse. But then after I had clicked around the UI a few times, the focus ring would stop showing up. That level of inconsistency made me think that the only reasonable solution is to use `:focus` to turn the focus ring on all the time and keep things consistent. 

My hunch was that this had something to do with JupyterLab's programmatic focus handling. Sure enough, I was able to reproduce this behavior (Chrome, Firefox, Safari) in a code pen: [.focus() on mousedown coerces :focus-visible](https://codepen.io/gabalafou-the-looper/pen/vYQxMVd).

Reply:
Nice! I think This looks definitely more subtle than the previous shades without making it unnoticeable and thus, defeating its purpose. 
I see no reason why this should not be merged, especially since the discussed alternatives (here) would probably cause more confusion among users and a focus ring is somewhat a more standard approach.

On the `focus-visible` note I've seen plenty of arguments against relying on this due to flaky/not consistent behaviours across web browsers.

Reply:
it would be great to merge this because the keyboard UX is really blocking jupyter from a keyboard only experience. 

when this closes. it is probably worth opening an issue that summarizes the focus styling that has been done so we don't repeat past experiments. having a working focus indicator will be good in getting feedback about how to improve that styling. utlimately, users should be able to customize these preferences because we won't find a one size fits all solution. 

it is probably work including https://www.sarasoueidan.com/blog/focus-indicators/ in references, this document thoroughly explores focus styling across browsers.